src/components/Reports/StaffExpenseReport/SettingsDialog/SettingsDialog.tsx

```diff
@@ -177,7 +229,11 @@ export const SettingsDialog: React.FC<SettingsDialogProps> = ({
                         endDate: false,
                       });
                     }
-                    setTimeout(() => validateForm(), 0);
+                    validateAndRefetch(validateForm, {
+                      ...values,
+                      selectedDateRange: value as DateRange,
```

Comment:
I think `value` can be `null`. Did you mean this?
```suggestion
                      selectedDateRange: value as (DateRange | null),
```